### PR TITLE
EE: Document key-auth support in on-prem (self-hosted) application registration, 2.2.1.0 patch release (READY FOR DOCS REVIEW)

### PR DESCRIPTION
**Pull request by tlblessing**
https://konghq.atlassian.net/browse/DOCS-1465 EE support for key-auth app reg
~~https://konghq.atlassian.net/browse/DOCS-1500 updates for key-auth plugin~~ will do for 2.3 CE release (Kong key-auth is open source version of plugin) on Jan 12 after this PR is merged. Note that keys already work in the query param or header; these new fields provide the ability to disallow those locations. key in body is set to false by default and is an already existing config parameter.

Direct review links:

- **NEW TOPIC** Enable key Auth for App Reg (Kong Mgr GUI) https://deploy-preview-2527--kongdocs.netlify.app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin/

- Enable App Reg (Kong Mgr GUI) updated prereqs and next steps for key auth: https://deploy-preview-2527--kongdocs.netlify.app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration/

- App reg plugin on hub added supported plugin for auth strategy: https://deploy-preview-2527--kongdocs.netlify.app/hub/kong-inc/application-registration/

image

- Auth Provider Strategy updated for key auth, renamed from index file so this shows up in TOC: https://deploy-preview-2527--kongdocs.netlify.app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy/

- Key Auth (open source) plugin, mention usage with app reg and add link, copy edits, added datatypes, fixed w and w/out db tabs, added missing example for request body existing param, ~~added 2 new fields: key_in_query, key_in_header (for 2.3.3.x only in a few weeks but they are already available in Kong Cloud? Can revert version/hide fields if so. Note internal plugin v is 2.4 yet community release is 2.3?~~)
 https://deploy-preview-2527--kongdocs.netlify.app/hub/kong-inc/key-auth/

- ODIC plugin, mention usage with app reg and add links https://deploy-preview-2527--kongdocs.netlify.app/hub/kong-inc/openid-connect/

image

Reviewers: Note that there are tickets in the backlog to add Kong Manager UI topics in this section for OIDC and OAuth2. The OIDC version would only hit the main fields to enter and not contain a table of parameters because there are so many. Most of the fields are autopopulated in the GUI.

### CODE REVIEW

File: app/_hub/kong-inc/key-auth/index.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 name: Key Authentication
 publisher: Kong Inc.
-version: 2.4.0
+version: 2.2.0
```
Comment (tlblessing) [inline app/_hub/kong-inc/key-auth/index.md]: Put back to the version as I found it but suspect there is a missing doc version

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+The key auth plugin uses the same Client ID as generated for the Kong OAuth2 plugin.
+You can use the same Client ID credential for a Service that has the OAuth2 plugin enabled.
+
+## Prerequisites
+
+* {{site.ee_product_name}} is installed, version 2.2.1.0 or later.
+* Create a Service.
+* Activate your application for a Service if you have not already done so. The
+Service Contract must be approved by an Admin if auto approve is not enabled.
+* Enable the [Application Registration plugin](/enterprise/{{page.kong_version}}/developer-portal/administration/application-registration/enable-application-registration) on a Service.
```
Comment (nateslo) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:19]: I think the order of these two bullet points should be swapped. Application Registration needs to be applied before a developer can request access to the service. 
```suggestion
* Enable the [Application Registration plugin](/enterprise/{{page.kong_version}}/developer-portal/administration/application-registration/enable-application-registration) on a Service.
* Activate your application for a Service if you have not already done so. The
Service Contract must be approved by an Admin if auto approve is not enabled.
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+```bash
+$ curl -X POST {proxy}/{route}?apikey={CLIENT_ID}
+```
+
+{% endnavtab %}
+{% navtab Using HTTPie %}
+
+```bash
+$ http :8000/{proxy}/{route}?apikey={CLIENT_ID}
```
Comment (nateslo) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:92]: I would remove the port here and on other httpie snippets
```suggestion
$ http {proxy}/{route}?apikey={CLIENT_ID}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+```bash
+$ curl -X POST {proxy}/{route} \
+--header "apikey: {CLIENT_ID}"
+```
+
+{% endnavtab %}
+{% navtab Using HTTPie %}
+
+```bash
+$ http :8000/{proxy}/{route} apikey:{CLIENT_ID}
```
Comment (nateslo) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:119]:
```suggestion
$ http {proxy}/{route} apikey:{CLIENT_ID}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+--data "apikey:={CLIENT_ID}"
+```
+
+**Note:** The `key_in_body` parameter must be set to `true`.
+
+{% endnavtab %}
+{% navtab Using HTTPie %}
+
+```bash
+$ http :8000/{proxy}/{route} apikey={CLIENT_ID}
```
Comment (nateslo) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:141]:
```suggestion
$ http {proxy}/{route} apikey={CLIENT_ID}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md [hunk 6]
```diff
@@ -3,21 +3,21 @@ title: Enable Application Registration
 ---
 
 ## Introduction
-Applications allow registered developers on Kong Developer Portal to
+Application Registration allows registered developers on the Kong Developer Portal to
 authenticate with OAuth2 against a Service on Kong. Either Kong or an
```
Comment (nateslo) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md:7]:
```suggestion
authenticate with supported Authentication
  plugins against a Service on Kong. Either Kong or an
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+The third-party authorization strategy (`external-oauth2`) applies to all
+applications across all Workspaces (Dev Portals) in a Kong cluster.
+
+### Authorization provider strategy configuration for Application Registration {#portal-app-auth}
+
+The `portal_app_auth` configuration option must be set in `kong.conf` to enable
+the Developer Portal Application Registration plugin with your chosen
+authorization strategy. The default setting (`kong-oauth2`) accommodates the Kong
+OAuth2 or Kong Key Authorization plugins.
```
Comment (nateslo) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:50]:
```suggestion
OAuth2 or Kong Key Authentication plugins.
```

**Review by nateslo — APPROVED** [review]
LGTM!

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+of their choice.
+
+In the {{site.ee_product_name}} 2.2.1.x version and later, support has been added
+for the Key Authentication plugin to use with Kong as the system of record.
+
+### Supported Authentication Plugins
+
+OAuth2 plugins for use with the Application Registration plugin:
+
+- When Kong is the system of record, the Application Registration plugin works
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:28]:
```suggestion
- When {{site.base_gateway}} is the system of record, the Application Registration plugin works
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+  in conjunction with the Kong OAuth2 or the Kong Key Authentication plugin.
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:29]:
```suggestion
  in conjunction with the OAuth2 or the Key Authentication plugin.
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+  <div class="alert alert-warning">
+    <strong>Important:</strong> The Kong OAuth2 plugin does not support
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:32]:
```suggestion
    <strong>Important:</strong> The OAuth2 plugin does not support
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+    hybrid mode. Since the Key Authentication plugin is using the `kong-oauth2` strategy
+    and client ID credential, hybrid mode is also not supported for the Key
+    Authentication plugin. If your organization uses hybrid mode, you must use an external identity
+    provider and configure the Kong OIDC plugin.
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:36]:
```suggestion
    provider and configure the OpenID Connect plugin.
```
I've noticed that we don't refer to this plugin as OIDC in the docs very often. Might be difficult to find if you use the short form only. If you'd like to keep using the term OIDC, then this should say "configure the OpenID Connect (OIDC) plugin".

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+  </div>
+
+- When an external OAuth2 is the system of record, the Application Registration
+  plugin works in conjunction with the Kong OIDC plugin.
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:40]:
```suggestion
  plugin works in conjunction with the OpenID Connect plugin.
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+
+The third-party authorization strategy (`external-oauth2`) applies to all
+applications across all Workspaces (Dev Portals) in a Kong cluster.
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:43]:
```suggestion
applications across all Workspaces (Dev Portals) in a {{site.base_gateway}} cluster.
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+
+### Authorization provider strategy configuration for Application Registration {#portal-app-auth}
+
+The `portal_app_auth` configuration option must be set in `kong.conf` to enable
+the Developer Portal Application Registration plugin with your chosen
+authorization strategy. The default setting (`kong-oauth2`) accommodates the Kong
+OAuth2 or Kong Key Authentication plugins.
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:50]:
```suggestion
authorization strategy. The default setting (`kong-oauth2`) accommodates the 
OAuth2 or Key Authentication plugins.
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+
+Available options:
+
+* `kong-oauth2`: Default. Kong is the system of record. The Application
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:54]:
```suggestion
* `kong-oauth2`: Default. {{site.base_gateway}} is the system of record. The Application
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+  Registration plugin is used in conjunction with the Kong OAuth2 or
+  Kong Key Authentication plugin. The `kong-oauth2` option can only be used with
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:56]:
```suggestion
  Registration plugin is used in conjunction with the OAuth2 or
  Key Authentication plugin. The `kong-oauth2` option can only be used with
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+  classic (traditional) deployments. Because the OAuth2 plugin requires a database
+  for every Kong instance, the `kong-oauth2` option cannot be used with hybrid mode
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:58]:
```suggestion
  for every gateway instance, the `kong-oauth2` option cannot be used with hybrid mode
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+  deployments.
+* `external-oauth2`: An external IdP is the system of record. The
+  Portal Application Registration plugin is used in conjunction with the Kong
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:61]:
```suggestion
  Portal Application Registration plugin is used in conjunction with the 
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+
+   ```
+   portal_app_auth = external-oauth2
+            # Developer Portal application registration
+            # auth provider and strategy. Must be set to configure
+            # authentication in conjunction with the application_registration plugin.
+            # Currently accepts kong-oauth2 or external-oauth2.
+   ```
+
+2. [Restart](https://docs.konghq.com/2.1.x/cli/#kong-restart) your Kong Enterprise
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:83]:
```suggestion
2. [Restart](/enterprise/{{page.kong_version}}/cli/#kong-restart) your Kong Enterprise
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+   instance.
+
+### Next Steps
+
+1. Enable the [Application Registration plugin](/enterprise/{{page.kong_version}}/developer-portal/administration/application-registration/enable-application-registration) on a Service.
+
+2. Enable a supported authentication plugin on the same Service as the Application Registration plugin,
+   as appropriate for your authorization strategy.
+
+    Kong OAuth2 strategy:
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:93]:
```suggestion
    {{site.base_gateway}} OAuth2 strategy:
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 21]
```diff
@@ -0,0 +1,123 @@
...
+
+    * If using the Kong-managed authorization strategy
+    (`kong-oauth2`) with the OAuth2 plugin, configure the Kong [OAuth2](/hub/kong-inc/oauth2/)
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:96]:
```suggestion
    (`kong-oauth2`) with the OAuth2 plugin, configure the [OAuth2](/hub/kong-inc/oauth2/)
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 22]
```diff
@@ -0,0 +1,123 @@
...
+    plugin on the same Service as the Application Registration plugin. You can use either the
+    Kong Manager GUI or cURL commands as documented on the [Plugin Hub](/hub/).
+    The OAuth2 plugin cannot be used in hybrid mode.
+    * If using the Kong-managed authorization strategy
+    (`kong-oauth2`) with key authentication, configure the Kong
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:101]:
```suggestion
    (`kong-oauth2`) with key authentication, configure the
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 23]
```diff
@@ -0,0 +1,123 @@
...
+### Next Steps
+
+1. Enable the [Application Registration plugin](/enterprise/{{page.kong_version}}/developer-portal/administration/application-registration/enable-application-registration) on a Service.
+
+2. Enable a supported authentication plugin on the same Service as the Application Registration plugin,
+   as appropriate for your authorization strategy.
+
+    Kong OAuth2 strategy:
+
+    * If using the Kong-managed authorization strategy
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:95]: Kong-managed implies that Kong the company manages the strategy. What you want to say is that Kong Gateway manages it, correct?

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 24]
```diff
@@ -0,0 +1,123 @@
...
+
+    2. If using the third-party authorization strategy
+    (`external-oauth2`), configure the OIDC plugin on the same Service as the
+    Application Registration plugin. You can use the Kong Manager GUI
+    or cURL commands as documented on the [Plugin Hub](/hub/kong-inc/openid-connect).
+    When your deployment is hybrid mode, the OIDC plugin must be configured to handle
+    authentication for the Portal Application Registration plugin.
+
+    3. Configure the identity provider for your application, configure your
+    application in Kong, and associate them with each other. See the
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:121]:
```suggestion
    application in {{site.base_gateway}}, and associate them with each other. See the
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md [hunk 25]
```diff
@@ -3,21 +3,21 @@ title: Enable Application Registration
 ---
 
 ## Introduction
-Applications allow registered developers on Kong Developer Portal to
-authenticate with OAuth2 against a Service on Kong. Either Kong or an
-external identity provider
-admins can selectively admit access to Services using Kong Manager.
+Application Registration allows registered developers on the Kong Developer Portal to
+authenticate with supported Authentication plugins against a Service on Kong. Either Kong or an
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md:7]:
```suggestion
authenticate with supported Authentication plugins against a Service on Kong. Either {{site.base_gateway}} or
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md [hunk 26]
```diff
@@ -46,56 +46,45 @@ In Kong Manager, access the Service for which you want to enable Application Reg
...
+| `Show Issuer` | Displays the Issuer URL in the Service Details page. Default: `false`. **Important:** Exposing the **Issuer URL** is essential for the [Authorization Code Flow](/enterprise/{{page.kong_version}}/developer-portal/administration/application-registration/3rd-party-oauth/#ac-flow) workflow configured for third-party identity providers. |
 
-Default: `false`
+## Next steps
 
-Displays the Issuer URL in the Service Details. **Note:** Exposing
-the Issuer URL is essential for the
-[Authorization Code Flow](/enterprise/{{page.kong_version}}/developer-portal/administration/application-registration/3rd-party-oauth/#ac-flow)
-workflow configured for third-party identity providers.
+Kong OAuth2 strategy:
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md:71]:
```suggestion
{{site.base_gateway}} OAuth2 strategy:
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md [hunk 27]
```diff
@@ -46,56 +46,45 @@ In Kong Manager, access the Service for which you want to enable Application Reg
...
 
-![Issuer URL](/assets/images/docs/dev-portal/dev-portal-issuer-url.png)
+* If using the Kong-managed authorization strategy
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md:73]: same comment as above, this should probably be Kong Gateway-managed. Applies everywhere.

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 28]
```diff
@@ -0,0 +1,145 @@
...
+![Services Pane](/assets/images/docs/dev-portal/portal-info-modal-key-auth.png)
+
+Scroll to view all of the available examples.
+
+![Service Details Page Embedded Key Usage Instructions](/assets/images/docs/dev-portal/service-details-key-auth-usage.png)
+
+### Make a request with the key as a query string parameter
+
+{% navtabs %}
+{% navtab Using cURL %}
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:82]:
```suggestion
{% navtab cURL %}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 29]
```diff
@@ -0,0 +1,145 @@
...
+
+![Services Pane](/assets/images/docs/dev-portal/portal-info-modal-key-auth.png)
+
+Scroll to view all of the available examples.
+
+![Service Details Page Embedded Key Usage Instructions](/assets/images/docs/dev-portal/service-details-key-auth-usage.png)
+
+### Make a request with the key as a query string parameter
+
+{% navtabs %}
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:81]:
```suggestion
{% navtabs codeblock %}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 30]
```diff
@@ -0,0 +1,145 @@
...
+{% navtab Using cURL %}
+
+```bash
+$ curl -X POST {proxy}/{route}?apikey={CLIENT_ID}
+```
+
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:87]: Remove blank lines around code example (83 and 86). Same goes for all subsequent other code examples in the {% navtabs codeblock %} style. 

(Can't add a suggestion, github gets confused by too many backticks in a row)

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 31]
```diff
@@ -0,0 +1,145 @@
...
+{% endnavtab %}
+{% navtab Using HTTPie %}
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:89]:
```suggestion
{% navtab HTTPie %}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 32]
```diff
@@ -0,0 +1,145 @@
...
+Response (will be the same for all valid requests regardless of key location):
+
+```bash
+HTTP/1.1 200 OK
+...
+```
+
+### Make a request with the key in a header
+
+{% navtabs %}
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:107]:
```suggestion
{% navtabs codeblock %}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 33]
```diff
@@ -0,0 +1,145 @@
...
+{% navtab Using cURL %}
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:108]:
```suggestion
{% navtab cURL %}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 34]
```diff
@@ -0,0 +1,145 @@
...
+
+```bash
+$ curl -X POST {proxy}/{route} \
+--header "apikey: {CLIENT_ID}"
+```
+
+{% endnavtab %}
+{% navtab Using HTTPie %}
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:116]:
```suggestion
{% navtab HTTPie %}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 35]
```diff
@@ -0,0 +1,145 @@
...
+```bash
+$ http {proxy}/{route} apikey:{CLIENT_ID}
+```
+
+{% endnavtab %}
+{% endnavtabs %}
+
+### Make a request with the key in the body
+
+{% navtabs %}
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:127]:
```suggestion
{% navtabs codeblock %}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 36]
```diff
@@ -0,0 +1,145 @@
...
+{% navtab Using cURL %}
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:128]:
```suggestion
{% navtab cURL %}
```

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 37]
```diff
@@ -0,0 +1,145 @@
...
+
+```bash
+$ curl -X POST {proxy}/{route} \
+--data "apikey:={CLIENT_ID}"
+```
+
+**Note:** The `key_in_body` parameter must be set to `true`.
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:135]: Move this line outside of the tabs, as it should apply to both cURL and HTTPie examples.

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md [hunk 38]
```diff
@@ -0,0 +1,145 @@
...
+
+{% endnavtab %}
+{% navtab Using HTTPie %}
```
Comment (lena-larionova) [inline app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md:138]:
```suggestion
{% navtab HTTPie %}
```

File: app/_hub/kong-inc/key-auth/index.md [hunk 39]
```diff
@@ -67,42 +75,59 @@ params:
...
+
+   ## Case sensitivity
+
+    Note that, according to their respective specifications, HTTP header names are treated as case _insensitive_, while HTTP query string parameter names are treated as case _sensitive_. Kong follows these specifications as designed, meaning that the `key_names` configuration values
+    are treated differently when searching the request header fields versus searching the query string. As a best practice, administrators
+    are advised against defining case-sensitive `key_names` values when expecting the authorization keys to be sent in the request headers.
 
     <div class="alert alert-warning">
-        <center>The option `config.run_on_preflight` is only available from version `0.11.1` and later</center>
+        <center>The option `config.run_on_preflight` is only available from versions `0.11.1` and later.</center>
```
[the remark this answers is not in the document]
Comment (tlblessing) [reply app/_hub/kong-inc/key-auth/index.md:126]: That note is so ancient and notes like that are all over the plugin docs and maybe elsewhere. This is Kong Gateway version.

File: app/_hub/kong-inc/key-auth/index.md [hunk 40]
```diff
@@ -131,27 +160,41 @@ consumers:
...
 service, you must add the new consumer to the allowed group. See
 [ACL: Associating Consumers][acl-associating] for details.
 
 ### Create a Key
 
-{% tabs %}
-{% tab With a database %}
-You can provision new credentials by making the following HTTP request:
+<div class="alert alert-warning">
+  <strong>Note:</strong> It is recommended to let Kong auto-generate the key. Only specify it yourself if
```
[the remark this answers is not in the document]
Comment (tlblessing) [reply app/_hub/kong-inc/key-auth/index.md:181]: This sounds odd.

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md [hunk 41]
```diff
@@ -0,0 +1,123 @@
+---
+title: Authorization Provider Strategy for Application Registration
+---
+
+## Overview
```
[the remark this answers is not in the document]
Comment (tlblessing) [reply app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:5]: I intend to add a new App Reg Overview topic to both on prem and Konnect. There are probably some customers who have been through the evolving security support in Kong Enterprise Gateway. I have a backlog ticket of course. This auth provider strategy and history of course is not applicable to Konnect. I'll reorg this but prefer to add this comment to another ticket so we can get this merged.

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md (hunk 23 above)
Comment (tlblessing) [reply app/enterprise/2.2.x/developer-portal/administration/application-registration/auth-provider-strategy.md:95]: yeah will rework

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md (hunk 27 above)
Comment (tlblessing) [reply app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md:73]: Well that seems to make it much longer than it needs to be. It is the 'kong-oauth2' strategy.

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md (hunk 26 above)
Comment (tlblessing) [reply app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-application-registration.md:71]: will rework. this rebrand is a bit unwieldy.

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md (hunk 37 above)
Comment (tlblessing) [reply app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md]: > Move this line outside of the tabs, as it should apply to both cURL and HTTPie examples.

Done in commit 0110381

File: app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md (hunk 30 above)
Comment (tlblessing) [reply app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin.md]: > Remove blank lines around code example (83 and 86). Same goes for all subsequent other code examples in the {% navtabs codeblock %} style.
> 
> (Can't add a suggestion, github gets confused by too many backticks in a row)

Done in commit 4eb4be6

File: app/_hub/kong-inc/key-auth/index.md (hunk 39 above)
[the remark this answers is not in the document]
Comment (tlblessing) [reply app/_hub/kong-inc/key-auth/index.md:126]: @GayleNeumann recommends removing these notes.

[the remark this answers is not in the document]
Comment (lena-larionova) [reply app/_hub/kong-inc/key-auth/index.md:126]: Seconded, that's super old info that should no longer be relevant.

**Comment by lena-larionova** [comment]
I agree with all of your comments on my suggestions. I should've checked the preview yesterday though - looks like the screenshots on https://deploy-preview-2527--kongdocs.netlify.app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin/ are blurry. Even when you click to expand, ie open to original size, the originals are blurry.

**Comment by tlblessing** [comment]
> @tlblessing I agree with all of your comments on my suggestions.
> 
> I should've checked the preview yesterday though - looks like the screenshots on https://deploy-preview-2527--kongdocs.netlify.app/enterprise/2.2.x/developer-portal/administration/application-registration/enable-key-auth-plugin/ are blurry. Even when you click to expand, ie open to original size, the originals are blurry.

@lena-larionova yes like I said to Nate: Some of the screenshots look blurry but I can get better ones after I have time to install the released version. Still working on that. Some of these were taken off of Nate's video demo, and that's why they look blurry. Desperate times, desperate measures. Will add to my follow-up ticket for app reg overview.

File: app/_hub/kong-inc/key-auth/index.md (hunk 39 above)
[the remark this answers is not in the document]
Comment (tlblessing) [reply app/_hub/kong-inc/key-auth/index.md]: Done in commit d147607

**Review by lena-larionova — APPROVED** [review]
LGTM, as long as screenshots and overview are fixed in subsequent PRs.

**Pull request merged**